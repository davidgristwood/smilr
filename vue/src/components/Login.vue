<template>
  <b-card border-variant="primary" header-bg-variant="primary" header-text-variant="white">
    <h1 slot="header">
      <fa icon="user" /> &nbsp; Login Required
    </h1>
    To access this part of the application you must be logged in
    <br><br>
    <p class="text-center">
      <b-button size="lg" variant="success" @click="login">
        &nbsp;&nbsp; <fa icon="user" /> LOGIN WITH AZURE ACTIVE DIRECTORY &nbsp;&nbsp;
      </b-button>
    </p>

    <b-alert v-if="loginFailed" show>
      <b>Login failed!</b> <br>{{ loginError }}
    </b-alert>
  </b-card>
</template>

<script>
import auth from '../mixins/auth'

export default {
  name: 'Login',

  mixins: [ auth ],

  props: {
    redir: {
      type: String,
      default: ''
    }
  },

  data() {
    return {
      loginFailed: false,
      loginError: null
    }
  },

  methods: {
    async login() {
      try {
        this.loginFailed = false
        await this.authLogin()

        if (!this.user() || !this.user().userName) {
          throw new Error('Login failed')
        }

        // Direct to root or redir path if provided
        if (this.redir) {
          this.$router.push({ name: this.redir })
        } else {
          this.$router.push({ name: 'home' }).catch(() => {})
        }
      } catch (err){
        console.log('### MSAL Error ' + err.toString())
        this.loginFailed = true
        this.loginError = err.toString()
      }
    }
  }
}
</script>