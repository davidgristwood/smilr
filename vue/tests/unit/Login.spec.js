import { mount, createLocalVue } from '@vue/test-utils'
import Login from '@/components/Login.vue'
import flushPromises from 'flush-promises'

import BootstrapVue from 'bootstrap-vue'
const localVue = createLocalVue()
localVue.use(BootstrapVue)

import VueRouter from 'vue-router'
localVue.use(VueRouter)
const router = new VueRouter()

describe('Login.vue', () => {
  it('renders login screen', async () => {
    const wrapper = mount(Login, { localVue, router })

    wrapper.find('button').trigger('click')
    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})
