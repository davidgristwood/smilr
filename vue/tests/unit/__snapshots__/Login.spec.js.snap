// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue renders login screen 1`] = `
<div class="card border-primary">
  <!---->
  <div class="card-header bg-primary text-white">
    <h1>
      <fa icon="user"></fa> &nbsp; Login Required
    </h1>
  </div>
  <div class="card-body">
    <!---->
    <!---->
    To access this part of the application you must be logged in
    <br><br>
    <p class="text-center"><button type="button" class="btn btn-success btn-lg">
        &nbsp;&nbsp; <fa icon="user"></fa> LOGIN WITH AZURE ACTIVE DIRECTORY &nbsp;&nbsp;
      </button></p>
    <!---->
  </div>
  <!---->
  <!---->
</div>
`;
